pkg/workloads/console/runner/runner.go

```diff
@@ -251,72 +252,101 @@ func (c *Runner) waitForSuccess(ctx context.Context, csl *workloadsv1alpha1.Cons
 	}
 
 	listOptions := metav1.SingleObject(pod.ObjectMeta)
-	w, err := c.clientset.CoreV1().Pods(pod.Namespace).Watch(ctx, listOptions)
-	if err != nil {
-		return fmt.Errorf("error watching pod: %w", err)
-	}
 
-	// We need to fetch the pod again now we have a watcher to avoid a race
-	// where the pod completed before we were listening for watch events
-	pod, _, err = c.GetAttachablePod(ctx, csl)
-	if err != nil {
-		// If we can't find the pod, then we should assume it finished successfully. Otherwise
-		// we might race against the operator to access a pod it wants to delete, and cause
-		// our runner to exit with error when all is fine.
-		//
-		// TODO: It may be better to recheck the console and look in its status?
-		if apierrors.IsNotFound(err) {
-			return nil
+	var rv string
```

Comment:
I don't think there's any need to initialise this as blank? We have an RV already in `pod.ObjectMeta.ResourceVersion`

Reply:
The status object has the latest `ResourceVersion` in it.

So I am initialising a seperate variable that we can use to overwrite the `ResourceVersion` field in the subsequent watches we do, as opposed to starting from scratch.

It also works for the second `break` to escape the `for` loop that reads the watch.

We set `rv` when we have received an expired status, that breaks out of the select, but we want to then break out of the `for` loop, because otherwise we risk looping forever.

It is late, and maybe I'm overcomplicating things, but that was my reasoning behind it.

Reply:
Oh will the `pod.ObjectMeta.ResourceVersion` update too? Then yeah, this is probably fine.

But will need to find a way to break from the `for` loop below. Could always use a label, but they just look ugly :)

Reply:
> So I am initialising a seperate variable that we can use to overwrite the ResourceVersion field in the subsequent watches we do, as opposed to starting from scratch.

But why not initialise this to `pod.ObjectMeta.ResourceVersion` that comes back from our call to `c.GetAttachablePod` near the start of the function?